### DISCUSSION TITLE
zh-cn/posts/2024/12/05/oneplus-ace-3-install-pixelos/

### DISCUSSION THREAD

**Discussion by giscus** [CyrusYip/blog-comments#83] [Comments]
# zh-cn/posts/2024/12/05/oneplus-ace-3-install-pixelos/

 注意事项 # 本文的操作可能会过时，建议参考文末的文章。 拿到手机要马上关闭自动更新，避免更新到不合适刷机的版本。 解锁 bootloader 重启后马上关掉自动更新。 解锁 bootloader 时会清除数据。如果确定要刷机，建议买到手机马上解锁，后面拿到 root 权限可以用 Neo Backup 备份应用数据。 PixelOS 安装完第一次后就清除数据并重启，然后在设置里再次安装系统。我在 recovery 安装两次 PixelOS 应该是错误操作。 名词解释 # 刷机：安装非操作系统 ROM：操作系统 bootloader：引导程序，需要先解锁它才能安装其他操作系统 root：最高权限 ocdt.img：每台一加手机特有的分区 persist.img：每台手机特有的分区 恢复系统教程 # 先看这篇恢复系统的教程：How To Guide - [12R/Ace 3] EDL DownloadTool to restore your device to OxygenOS/ColorOS | XDA Forums，最好用不上啦。

https://cyrusyip.org/zh-cn/posts/2024/12/05/oneplus-ace-3-install-pixelos/

**Comment by Liam-Zhong** [comment]
>刷机：安装非操作系统
>
>ROM：操作系统

第一个是不是那些完整，第二个不是一种存储咩

**Reply by Liam-Zhong** [reply]
可恶，没搞懂这个评论的用法我再测试一下
> hello cyrusyip

从来没刷过机，iOS 不方便折腾

**Reply by CyrusYip** [reply]
感谢评论，我修改了刷机和 ROM 的定义。刷机定义一开始我写的是「安装非原装操作系统」，一下没注意只删除了「原装」二字。你对 ROM 的理解正确，不过在刷机的语境下 ROM 确实是指操作系统，这算是刷机界的术语吧。

如果你想玩刷机，可以买一台二手 Google Pixel，也可以看看网上的视频做个云玩家，哈哈。据我了解，Google Pixel 系列是最方便刷机的手机，ROM 也很多。注意要买无网络锁的版本，不然无法解锁 bootloader。

**Reply by Liam-Zhong** [reply]
哦哦原来如此~
>可以买一台二手 Google Pixel

有机会去搞台来体验下谷歌生态嘿嘿

**Reply by rxliuli** [reply]
应该补充一下 boot-aston-20241019.img 这里的文件来源，也是需要下载得到，吾辈是命令执行失败了才去找到这个链接
https://sourceforge.net/projects/pixelos-releases/files/fourteen/aston/recovery/

**Reply by CyrusYip** [reply]
我在文章补充了点击 Download 下载文件。

**Comment by rxliuli** [comment]
> 现在屏幕显示 Active slot: b 和 Do you want to reboot to recovery now?，选择 Yes。

> 现在屏幕显示 Active slot: a，选择 Apply update -> Apply from ADB，再次安装 PixelOS（文档写要安装两次，这里是错误操作，应该是进入系统后再次安装）。

这儿前面写如果是不是更好？

**Reply by rxliuli** [reply]
> 选择 Factory reset -> Format data/factory reset -> Format data，返回，选择 Reboot system now。可以开机。

这一步之后就变砖了，可能兼容性有什么问题？

**Reply by CyrusYip** [reply]
刷机前的系统版本正确吗？

要求：OxygenOS(12R)/ColorOS(ACE3) version 14.0.0.810 or higher, but below OxygenOS 15.

**Reply by rxliuli** [reply]
Version number

PJE110_14.0.0.801(CN01)

Baseband version

Q_V1_P14,Q_V1_P14

Kernel version

5.15.123-android13-8-00758-9835dd0ff1fbc

**Reply by CyrusYip** [reply]
~~版本看起来没啥问题啊。我好像不应该把错误操作写到文章里，可能把你搞迷糊了。~~

你的版本过低了。

**Reply by CyrusYip** [reply]
你从头刷机试试。Do you want to reboot to recovery now? 这里选 NO。然后选择 Factory reset -> Format data/factory reset -> Format data，返回，选择 Reboot system now。

**Reply by rxliuli** [reply]
额，已经变砖，吾辈先看看怎么恢复到 ColorOS

**Reply by CyrusYip** [reply]
https://t.me/infernos_hideout

这个是维护者的群，可以去问问。

**Reply by CyrusYip** [reply]
我手机已经装好 Pixel OS 用着了，不方便重装一遍来修正本文，我最后决定删掉安装流程了。祝你救砖成功。

**Reply by CyrusYip** [reply]
下午看错了。你 ColorOS 版本过低了。

**Reply by rxliuli** [reply]
抱歉，确认是版本问题，安装流程有一点迷惑但应该没问题，810 和 801，我也没注意到

**Reply by CyrusYip** [reply]
@rxliuli 你最后装好 Pixel OS 啦？很好奇怎么救砖的。

**Reply by rxliuli** [reply]
付费救砖（另外 800 版本而非 810 版本安装 pixelos 其实也没问题，可能不同版本对应的实际操作不一样，只有 810 对应的操作才是和你完全一样的，吾辈实际上 sideload 了 4 次而非 2 次，不清楚有什么区别。

**Reply by CyrusYip** [reply]
相比于 LineageOS 的文档，Pixel OS 的安装文档写得太简洁了。我都不确定自己做得对不对。

[XDA 帖子](https://xdaforums.com/t/12r-ace3-rom-14-official-pixelos-aosp-19-10-24.4662225/)有这么一句话：

> If you are already on or above V14.0.0.700, flash the same update again
this is necessary so that both slots have same firmware in them
(Ignoring this step can get you bricked in future)

我挺担心下次升级的时候变砖了。我可能下次升级前先用 [DataBackup](https://github.com/XayahSuSuSu/Android-DataBackup) 或者 [Neo Backup](https://github.com/NeoApplications/Neo-Backup) 备份应用。

---

付费救砖多少钱啊？可能我有天也用得上。如果不方便可以不说。

**Reply by rxliuli** [reply]
付费救砖 + 刷入 pixelos 总共 150，之前单问刷入 pixelos 130，不清楚单独救砖花多少，但不可能只要 20 😂

**Reply by CyrusYip** [reply]
@rxliuli 最近安装了 PixelOS 15 并且更新了一次小版本。终于搞懂安装要求了，原来是安装两次正确版本的原厂系统（ColorOS），确保两个系统槽位都有正确的固件，PixelOS 在 recovery 安装一次就好。

**Reply by Youxikong** [reply]
> Version number
> 
> PJE110_14.0.0.801(CN01)
> 
> Baseband version
> 
> Q_V1_P14,Q_V1_P14
> 
> Kernel version
> 
> 5.15.123-android13-8-00758-9835dd0ff1fbc

底包不对，需要ColorOS 15以上的。

**Reply by Youxikong** [reply]
> @rxliuli 最近安装了 PixelOS 15 并且更新了一次小版本。终于搞懂安装要求了，原来是安装两次正确版本的原厂系统（ColorOS），确保两个系统槽位都有正确的固件，PixelOS 在 recovery 安装一次就好。

我上PixelOS的时候，找不到ColorOS的卡刷包，没有再刷一次相同版本，冒险刷了PixelOS。大佬你知道现在我是否可以线刷回ColorOS吗？

**Reply by CyrusYip** [reply]
@Youxikong  你现在用什么版本的 PixelOS？之前怎么安装的？

**Reply by CyrusYip** [reply]
> https://github.com/CyrusYip/blog-comments/discussions/83#discussioncomment-12230580

@Youxikong  他之前问的时候 PixelOS 版本还是 14，那时候是用 ColorOS 14 的。

**Reply by Youxikong** [reply]
> @Youxikong  你现在用什么版本的 PixelOS？之前怎么安装的？

我之前用的PixelOS 14，先edl模式，线刷回ColorOS 14，然后更新到ColorOS 15，接下来应该刷下同版本的官方ROM，但是我找不到资源，冒险上了PixelOS 15。

**Reply by CyrusYip** [reply]
@Youxikong 

我知道的安装 ColorOS 的方法就两种：fastboot、EDL。

Fastboot 刷机工具：

- [Fastboot Firmware Flasher](https://t.me/gt3neo5hub/521/232181)（从 PixelOS 14 回 ColorOS 14 的时候用过，没成功，不知道是不是操作失误）
- 在酷安搜「自信の靓仔」，他也有 fastboot 刷机工具（我没用过）

升级到 ColorOS 15 之后， EDL 工具就不能免授权使用了。

你可以试试这篇新教程：[Fix Sahara Communication/Protocol Failed Error in Oppo Flash Tool - DroidWin](https://droidwin.com/fix-sahara-communication-failed-error-in-oppo-flash-tool/)，回到 ColorOS 就安装两次 ColorOS 15.0.0.500（下载链接在[将一加 Ace 3 从 PixelOS 14 升级到 PixelOS 15](https://cyrusyip.org/zh-cn/posts/2025/02/13/oneplus-ace3-from-pixelos14-to-pixelos15/)）。如果用那篇教程搞不定，可以考虑付费购买授权码（我看闲鱼上有）、付费远程恢复系统、去售后中心。

**Reply by Youxikong** [reply]
@CyrusYip 我想想，卡刷ColorOS同版本包，是为了里面的一些分区镜像，让ab分区镜像一致。我下次试试看，提取卡刷包的payload.bin里的镜像，用fastboot刷入，在用安装PixelOS的流程安装PixelOS，试试能否成功。

**Reply by CyrusYip** [reply]
@Youxikong 我今天看到一篇用 fastboot 刷 ColorOS 15 的教程：[Flashing stock firmware on 12r/Ace 3 from fastboot.](https://xdaforums.com/t/flashing-stock-firmware-on-12r-ace-3-from-fastboot.4719377/)，你可以试试。

**Reply by Youxikong** [reply]
@CyrusYip 感觉和我思路差不多，但是应该有些分区必须要fastbootd模式才能刷入。我记得之前看到miui的线刷脚本就是这样的。

**Comment by CyrusYip** [comment]
以下是删去的 Pixel OS 安装流程，只是留作记录。

<details>
  <summary>点击展开内容</summary>


```shell
adb reboot bootloader
fastboot flash boot boot-aston-20241019.img
fastboot flash vendor_boot vendor_boot-aston-20241019.img
fastboot flash dtbo dtbo-aston-20241019.img
fastboot flash recovery recovery-aston-20241019.img
```

按音量键切换到 Recovery mode，按电源键确认。选择 Apply update -> Apply from ADB。

```
adb sideload PixelOS_aston-14.0-20241018-1605.zip
```

现在屏幕显示 `Active slot: b` 和 `Do you want to reboot to recovery now?`，选择 `Yes`。

现在屏幕显示 `Active slot: a`，选择 Apply update -> Apply from ADB，再次安装 PixelOS（文档写要安装两次，这里是错误操作，应该是进入系统后再次安装）。

```shell
adb sideload PixelOS_aston-14.0-20241018-1605.zip
```

这次报错了，先不管。

选择 Factory reset -> Format data/factory reset -> Format data，返回，选择 Reboot system now。可以开机。

开机连接 WIFI 后等了有两三分钟，显示 Slow connection，选择左边的 Set up offline。字体有点小，在设置里调大 Display size。

原系统在 https://resolution-viewer.cyrusyip.org/ 看到的缩放值好像是 350%

下拉通知栏，点击「Charging this device via USB」，选择 File Transfer。在电脑把 `PixelOS_aston-14.0-20241018-1605.zip` 复制到 Download 文件夹。

在手机打开 Settings -> System -> System updates -> Local update，选择刚刚的 `.zip`文件，点击 INSTALL，完成后点 Reboot，重启成功。

再重启一次，没事，应该算成功安装 PixelOS 了。

</details>

**Comment by Youxikong** [comment]
大佬，请教一个问题，我没有备份odct和persist分区，直接刷了pixel14，这个时候再备份这两个分区，还有用吗？

**Reply by CyrusYip** [reply]
我刚刚在 PixelOS_aston-14.0-20241018-1605 这个版本备份了 odct 和 persist，odct 和我在 ColorOS 提取的一样，persist 不一样。我解压了 PixelOS 的刷机包，发现里面也没有 odct 和 persist，所以刷入它应该不会改变 odct 和 persist，不知道为什么这里 persist 发生了变化。

推荐你还是备份这两个分区的，应该有用的，记得记下刷机前的原厂系统版本和目前的系统版本。

> 直接刷了pixel13

你这个版本是不是写错了？我看一加 Ace 3 的 PixelOS 只有 14 和 15。

**Reply by Youxikong** [reply]
确实，笔误了。我用payload-dumper查看，pixel 14的包是有这两个分区的。我试试看，有没有用。

顺便咨询下，如果因为这两个分区有问题，刷原厂包会不会导致砖了？

**Reply by CyrusYip** [reply]
我用 payload-dumper-go 看了，倒是没看到这两个分区。

```
boot.img
dtbo.img
init_boot.img
odm.img
product.img
recovery.img
system.img
system_dlkm.img
system_ext.img
vbmeta.img
vbmeta_system.img
vbmeta_vendor.img
vendor.img
vendor_boot.img
vendor_dlkm.img
```

> 顺便咨询下，如果因为这两个分区有问题，刷原厂包会不会导致砖了？

这个我也不清楚，我是看到不少帖子都推荐备份这两个分区，我就跟着做了。

**Reply by Youxikong** [reply]
万幸，还来得及备份，谢谢大佬。

**Comment by CyrusYip** [comment]
PixelOS 大版本升级流程来啦：[将一加 Ace 3 从 PixelOS 14 升级到 PixelOS 15 | 叶寻的博客](https://cyrusyip.org/zh-cn/posts/2025/02/13/oneplus-ace3-from-pixelos14-to-pixelos15/)。
